### ISSUE TITLE
Best way to support record only configuration

### ISSUE THREAD

**Issue by codeboten** [open-telemetry/opentelemetry-configuration#83]
The OTel specification allows users to set a [recording only](https://github.com/open-telemetry/opentelemetry-specification/blob/700f5138f4b10c6e9d11b41153f6e164c61f581d/specification/trace/sdk.md?plain=1#L343) option for sampler configuration.  This option is not available via environment variables today: https://github.com/open-telemetry/opentelemetry-specification/blob/700f5138f4b10c6e9d11b41153f6e164c61f581d/specification/configuration/sdk-environment-variables.md#general-sdk-configuration

Should this use-case be supported by the configuration file? If so would it be as simple as adding one more type of sampler here: https://github.com/open-telemetry/opentelemetry-configuration/blob/6215d43ee00c4ebe32ca8cbaf4c2807600b73945/schema/tracer_provider.json#L57

**Comment by jack-berg** [comment]
I don't think we should support this directly. We should focus on allowing samplers to be configured, both built in and custom implementations. The samplers we directly reference in the schema are defined in [Built-in sampler](https://github.com/open-telemetry/opentelemetry-specification/blob/700f5138f4b10c6e9d11b41153f6e164c61f581d/specification/trace/sdk.md?plain=1#L361). If there were to be one which was "AlwaysRecordOnly" or something like that, we ought to support it. 

Short of that, the way to use file configuration to get a sampler which returns `RECORD_ONLY` is to implement a custom sampler which returns the response, ensure its registered with an associated [component provider](https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/configuration/file-configuration.md#register-component-provider), and reference it by name in the configuration file.

**Comment by jack-berg** [comment]
There's a spec proposal to add a built-in sampler for this purpose: https://github.com/open-telemetry/opentelemetry-specification/pull/4699

**Comment by dlanov** [comment]
Hi, I’d like to work on this issue.

Now that the OpenTelemetry specification defines `AlwaysRecord` as a built-in stable sampler, I plan to add it to `schema/tracer_provider.yaml` with its required wrapped `root` sampler, add a validating example or snippet, regenerate the schema and documentation, and run the full repository validation.

Please let me know if this scope matches the expected implementation.
